### ISSUE TITLE
How to execute the tasks parallel in separate thread ?

### ISSUE THREAD

**Issue by newdev7** [BoltsFramework/Bolts-iOS#105]
I have 10 elements in an array. After checking each element in that array, I created the task. Now I want to run all the tasks parallel in different thread not in the main thread. For that I used the BFExecutor.
```
BFExecutor* executor = [BFExecutor executorWithBlock:^void(void(^block)()) {
                        dispatch_async(dispatch_get_global_queue(0,0), block);
                    }];
BFTask* task = [BFTask taskWithResult: nil];

[[task continueWithExecutor:executor withBlock:^id(BFTask* task) {
   return task2;
}] continueWithBlock:^id(BFTask* task) {
   # Do something with that task.result
   return nil;
}];
```

Is this is the correct way of using BFTask in separate thread ?
Thanks.

**Comment by nlutsenko** [comment]
Since you are using an executor with a global dispatch queue - you can create 10 separate tasks with the same executor and then wrap it with another task that will be the result of your task.
(also you can create executor directly with a dispatch queue, instead of wrapping in a block).
The code for this would look roughly like this:
```objc
    NSMutableArray *tasks = [NSMutableArray array];
    BFExecutor *executor = [BFExecutor executorWithDispatchQueue:dispatch_get_global_queue(DISPATCH_QUEUE_PRIORITY_DEFAULT, 0)];
    for (...) {
        BFTask *task = [BFTask taskFromExecutor:executor withBlock:^id{
            // do some work
            return nil; // return actual result
        }];
        [tasks addObject:task];
    }
    BFTask *finalTask = [BFTask taskForCompletionOfAllTasks:tasks];
    [finalTask continueWithBlock:^id(BFTask *task) {
        // All 10 tasks completed at this point, in parallel.
        return nil;
    }];
```

Hope this helps, let me know if there is anything else I can help you with.
